extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -40,41 +59,126 @@ AdditionalBeanBuildItem beans() {
     }
 
     @BuildStep
-    List<ReflectiveClassBuildItem> reflectiveClasses() {
+    NativeImageProxyDefinitionBuildItem connectionProxy(QuartzBuildTimeConfig config) {
+        if (config.storeType.equals(StoreType.DB)) {
+            return new NativeImageProxyDefinitionBuildItem(Connection.class.getName());
+        }
+        return null;
+    }
+
+    @BuildStep
+    QuartzJDBCDriverDialectBuildItem driver(Optional<DataSourceDriverBuildItem> dataSourceDriver,
+            QuartzBuildTimeConfig config) {
+        if (config.storeType == StoreType.RAM) {
+            if (config.clustered) {
+                throw new ConfigurationError("Clustered jobs configured with unsupported job store option");
+            }
+
+            return new QuartzJDBCDriverDialectBuildItem(Optional.empty());
+        }
+
+        if (!dataSourceDriver.isPresent()) {
+            String message = String.format(
+                    "JDBC Store configured but '%s' datasource is not configured properly. You can configure your datasource by following the guide available at: https://quarkus.io/guides/datasource-guide",
+                    config.dataSourceName.isPresent() ? config.dataSourceName.get() : "default");
+            throw new ConfigurationError(message);
+        }
+
+        return new QuartzJDBCDriverDialectBuildItem(Optional.of(guessDriver(dataSourceDriver)));
+    }
+
+    private String guessDriver(Optional<DataSourceDriverBuildItem> dataSourceDriver) {
+        if (!dataSourceDriver.isPresent()) {
+            return StdJDBCDelegate.class.getName();
+        }
+
+        String resolvedDriver = dataSourceDriver.get().getDriver();
+        if (resolvedDriver.contains("postgresql")) {
+            return PostgreSQLDelegate.class.getName();
+        }
+        if (resolvedDriver.contains("org.h2.Driver")) {
+            return HSQLDBDelegate.class.getName();
+        }
+
+        if (resolvedDriver.contains("com.microsoft.sqlserver.jdbc.SQLServerResource")) {
```

Comment:
Outch, this is a bit ugly. But we can refine later.

Reply:
Got you 👍